### PR TITLE
login: non existing user email error fix

### LINKED ISSUE #3309: access: InvenioWebAccessMailCookieError for non existing user email
Steps to reproduce:
1. Go to sign in panel
2. Click `Lost your password?`
3. Provide some non existing email

```
Traceback (most recent call last):
 File "/opt/zenodo/src/invenio/invenio/ext/legacy/__init__.py", line 124, in __call__
   response = self.app.full_dispatch_request()
 File "/opt/zenodo/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
   rv = self.handle_user_exception(e)
 File "/opt/zenodo/lib/python2.7/site-packages/flask_restful/__init__.py", line 270, in error_router
   return original_handler(e)
 File "/opt/zenodo/src/invenio/invenio/base/wrappers.py", line 125, in handle_user_exception
   return super(Flask, self).handle_user_exception(e)
 File "/opt/zenodo/lib/python2.7/site-packages/flask/app.py", line 1<*****>81, in handle_user_exception
   reraise(exc_type, exc_value, tb)
 File "/opt/zenodo/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
   rv = self.dispatch_request()
 File "/opt/zenodo/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
   return self.view_functions[rule.endpoint](**req.view_args)
 File "/opt/zenodo/src/invenio/invenio/modules/accounts/views/accounts.py", line <*****>05, in lost
   if reset_password(request.values['email'], g.ln):
 File "/opt/zenodo/src/invenio/invenio/ext/login/__init__.py", line 56, in reset_password
   days=CFG_WEBSESSION_RESET_PASSWORD_EXPIRE_IN_DAYS))
 File "/opt/zenodo/src/invenio/invenio/modules/access/mailcookie.py", line 71, in mail_cookie_create_pw_reset
   "Email '%s' doesn't exist" % email)
InvenioWebAccessMailCookieError: Email 'shirley@ibt.unam.mx' doesn't exist
```

The problem is in this line: https://github.com/inveniosoftware/invenio/blob/master/invenio/modules/access/mailcookie.py#L68

### PR DESCRIPTION
**Pull request by bouzlibop**
* FIX Provides flash message to indicate that an email with password
  recovery could not be sent. (closes #3309)

Signed-off-by: Adrian Pawel Baran <adrian.pawel.baran@cern.ch>

### CODE REVIEW

**Comment by bouzlibop** [comment]
Would it be enough if we provide flash message in this legacy code?

**Comment by jirikuncar** [comment]
@bouzlibop please amend the commit message:
- add FIX label
- finish sentence before mentioning issue

Example:
```
* FIX Provides flash message to indicate problem with creating token
   for password reset. (closes #3309)
```

File: invenio/ext/login/__init__.py [hunk 1]
```diff
@@ -41,14 +41,20 @@ def login_user(user, *args, **kwargs):
...
         ln = g.ln
     from invenio.modules.access.mailcookie import mail_cookie_create_pw_reset
-    reset_key = mail_cookie_create_pw_reset(email, cookie_timeout=timedelta(
-        days=CFG_WEBSESSION_RESET_PASSWORD_EXPIRE_IN_DAYS))
+    from invenio.modules.access.errors import InvenioWebAccessMailCookieError
+    try:
+        reset_key = mail_cookie_create_pw_reset(email, cookie_timeout=timedelta(
+            days=CFG_WEBSESSION_RESET_PASSWORD_EXPIRE_IN_DAYS))
+    except InvenioWebAccessMailCookieError:
+        flash(_("Provided URL is not valid"), "error")
```
Comment (jirikuncar) [inline invenio/ext/login/__init__.py]: IMHO it's not "URL" what it's invalid but "email address" or "data" in general.

**Comment by bouzlibop** [comment]
@jirikuncar, I've just updated commit msg for this one.

**Pull request merged**